packages/apputils-extension/src/index.ts

```diff
@@ -335,24 +335,36 @@ export const toggleHeader: JupyterFrontEndPlugin<void> = {
 async function updateTabTitle(workspace: string, db: IStateDB, name: string) {
   const data: any = await db.toJSON();
   let current: string = data['layout-restorer:data']?.main?.current;
-  if (current === undefined) {
+  // File name from current path
+  let currentFile: string | undefined = current
+    ?.split(':')[1]
+    ?.split('\\')
+    ?.pop()
+    ?.split('/')
+    ?.pop();
```

Comment:
Please try to use coreutils `PathExt.basename` (`import { PathExt } from '@jupyterlab/coreutils';`), this should simplify the code and make it more robust to some edge cases.

Reply:
Yes, this seems to be a better way around. I'll make changes accordingly and push here.

Reply:
I've made the necessary changes as mentioned.